auth_service/user_management/claims_repository/core/claims.py

```diff
@@ -37,17 +43,40 @@ def is_valid_claim(claim: Claim, now: Callable[[], DateTimeUTC] = now_as_utc) ->
     return not claim.revocation_date and claim.valid_from <= now() <= claim.valid_until
 
 
+def is_internal_claim(claim: Claim, visa_type: VisaType) -> bool:
+    """Check whether this is a valid internal claim of the given type."""
+    return (
+        claim.visa_type == visa_type
+        and claim.source == INTERNAL_SOURCE
+        and claim.asserted_by not in (None, AuthorityLevel.SELF)
+        # conditions are currently not supported,
+        # so if they exist the claim must be considered invalid
+        and not claim.conditions
+    )
+
+
 def is_data_steward_claim(claim: Claim) -> bool:
     """Check whether the given claim asserts a data steward role."""
-    if (
-        claim.visa_type != VisaType.GHGA_ROLE
-        or claim.source != INTERNAL_SOURCE
-        or not claim.asserted_by
-        or claim.asserted_by == AuthorityLevel.SELF
-    ):
+    if not is_internal_claim(claim, VisaType.GHGA_ROLE):
         return False
     visa_value = claim.visa_value
     if not isinstance(visa_value, str):
         return False
     role_name = visa_value.split("@", 1)[0]
     return role_name == DATA_STEWARD_ROLE
+
+
+def get_value_for_dataset(dataset_id: str) -> str:
+    """Return the Visa URL Claim for the given dataset."""
+    return f"{INTERNAL_SOURCE}/datasets/{dataset_id}"
```

Comment:
I don't think we will have URLs for datasets. What is wrong with the dataset id?

Reply:
Values must be URLs (see above) if we want to issue that claim as a valid Visa. So I'm just creating artificial URLs.

Reply:
Ah OK, didn't know that this is a requirement.